### PR TITLE
optimize button

### PR DESCRIPTION
**Pull request by nutty898**

### CODE REVIEW

File: cocos/gui/UIButton.cpp [hunk 1]
```diff
@@ -29,10 +29,10 @@
...
 namespace gui {
 
-#define NORMALRENDERERZ (0)
-#define PRESSEDRENDERERZ (0)
-#define DISABLEDRENDERERZ (0)
-#define TITLERENDERERZ (1)
+#define NORMALRENDERERZ (-2)
+#define PRESSEDRENDERERZ (-2)
+#define DISABLEDRENDERERZ (-2)
+#define TITLERENDERERZ (-1)
```
Comment (dumganhar) [inline cocos/gui/UIButton.cpp]: I think using `static const int XXX = 111;` will be better. Right?

File: cocos/gui/UIButton.h [hunk 2]
```diff
@@ -132,16 +132,16 @@ class Button : public Widget
     virtual void setScale9Enabled(bool able);
     
     //override "setFlipX" of widget.
-    virtual void setFlipX(bool flipX) override;
+    virtual void setFlipX(bool flipX);
     
     //override "setFlipY" of widget.
-    virtual void setFlipY(bool flipY) override;
+    virtual void setFlipY(bool flipY);
```
Comment (dumganhar) [inline cocos/gui/UIButton.h]: Why to remove `override` keyword?

File: cocos/gui/UIButton.cpp (hunk 1 above)
Comment (minggo) [inline cocos/gui/UIButton.cpp]: Yep.
And it is better to use `_` to connect multiple words with constants, such as

```
NORMALRENDERERZ -> NORMAL_RENDERER_Z
```

Comment (dumganhar) [inline cocos/gui/UIButton.cpp]: +1 :)

Comment (nutty898) [inline cocos/gui/UIButton.cpp]: OK

**Pull request merged**